### PR TITLE
Add `base_flow` and `gas_molar_mass` when writing MATGAS files.

### PR DESCRIPTION
**Pull request by tasseff**
Maybe there's a reason for not including these in the MATGAS export, but I was having issues generating feasible new scenarios using the I/O functions. Infeasibility seemed to be caused by the exported files lacking `base_flow`.

### CODE REVIEW

File: src/io/matgas.jl [hunk 1]
```diff
@@ -292,9 +292,9 @@ function parse_m_string(data_string::String)
         case["source_version"] = "0.0.0+"
     end
 
-    required_metadata_names = ["mgc.gas_specific_gravity", "mgc.specific_heat_capacity_ratio", "mgc.temperature", "mgc.compressibility_factor", "mgc.units"]
+    required_metadata_names = ["mgc.gas_specific_gravity", "mgc.gas_molar_mass", "mgc.specific_heat_capacity_ratio", "mgc.temperature", "mgc.compressibility_factor", "mgc.units"]
```
Comment (kaarthiksundar) [inline src/io/matgas.jl:295]: Why is gas_molar_mass both in required and optional meta data names.

**Comment by kaarthiksundar** [comment]
> Maybe there's a reason for not including these in the MATGAS export, but I was having issues generating feasible new scenarios using the I/O functions. Infeasibility seemed to be caused by the exported files lacking `base_flow`.

`base_flow` is something that can be calculated using base pressure and length. This was a requirement in the old versions because base_flow and base_pressure were being used for non-dimensionalization. So, I don't clearly understand why it would cause feasibility issues.

File: src/io/matgas.jl (hunk 1 above)
Comment (tasseff) [reply src/io/matgas.jl]: Sorry, this was a mistake. Should be fixed, now.

**Comment by tasseff** [comment]
Here's a short example that will result in infeasibility on the second solve with the current v0.7-rc branch. I'm not familiar enough with the code to know where the infeasibility originates, only that inclusion of `mgc.base_flow = 604;` in the generated `tmp.m` will result in feasibility.

File: src/io/matgas.jl [hunk 2]
```diff
@@ -693,6 +703,7 @@ const _units = Dict{String,Dict{String,String}}(
         "sound_speed" => "m/s",
         "base_pressure" => "Pa",
         "base_length" => "m",
+        "base_flow" => "kg/m^2/s", # TODO: Is this correct?
```
Comment (kaarthiksundar) [inline src/io/matgas.jl:706]: base_flow's units is kg/s. base_flux in kg/m^2/s

File: src/io/matgas.jl [hunk 3]
```diff
@@ -703,11 +714,12 @@ const _units = Dict{String,Dict{String,String}}(
         "sound_speed" => "m/s",
         "base_pressure" => "psi",
         "base_length" => "miles",
+        "base_flow" => "lb/ft^2/s", # TODO: Is this correct?
```
Comment (kaarthiksundar) [inline src/io/matgas.jl:717]: Same issue as in the SI case

**Comment by kaarthiksundar** [comment]
> Here's a short example that will result in infeasibility on the second solve with the current v0.7-rc branch. I'm not familiar enough with the code to know where the infeasibility originates, only that inclusion of `mgc.base_flow = 604;` in the generated `tmp.m` will result in feasibility.
> 
> ```julia
> import Cbc
> import GasModels
> import Ipopt
> import JuMP
> import Juniper
> 
> cbc = JuMP.optimizer_with_attributes(Cbc.Optimizer, "logLevel"=>0)
> ipopt = JuMP.optimizer_with_attributes(Ipopt.Optimizer, "tol"=>1.0e-4, "print_level"=>0, "sb"=>"yes")
> juniper = JuMP.optimizer_with_attributes(Juniper.Optimizer, "nl_solver"=>ipopt, "mip_solver"=>cbc)
> 
> # Original MATGAS file.
> GasModels.run_gf("test/data/matgas/gaslib-40.m", GasModels.MISOCPGasModel, juniper)
> 
> # Exported MATGAS file.
> data = GasModels.parse_file("test/data/matgas/gaslib-40.m")
> GasModels.write_matgas!(data, "tmp.m")
> GasModels.run_gf("tmp.m", GasModels.MISOCPGasModel, juniper)
> ```

I will look into this a little more carefully. Can you hold on for a day or two.

**Comment by rb004f** [comment]
Yes, a number of these problems appear to be very sensitive to the choice of the normalization constant used for flow (for the most part base_flow = max_flow seems to work).  We'll want to spend some time diving into these problems to try understand the source of the numerical challenges, but a manual choice of base_flow ensures the tests will pass.  Currently, `gas_molar_mass` is only needed as field a when `sound_speed` is not provided... it is used to compute `sound_speed` in that case. But good to preserve the data.

**Comment by tasseff** [comment]
Think I've figured this out. Will submit a different pull request shortly.

**Pull request closed without merging**